docs/reference/search/rrf.asciidoc

```diff
@@ -1,6 +1,423 @@
+                "type": "dense_vector",
+                "dims": 1,
+                "index": true,
+                "similarity": "l2_norm"
+            },
+            "integer" : {
+                "type" : "integer"
+            }
+        }
+    }
+}
+
+PUT example-index/_doc/1
+{
+    "text" : "rrf",
+    "vector" : [5],
+    "integer": 1
+}
+
+PUT example-index/_doc/2
+{
+    "text" : "rrf rrf",
+    "vector" : [4],
+    "integer": 2
+}
+
+PUT example-index/_doc/3
+{
+    "text" : "rrf rrf rrf",
+    "vector" : [3],
+    "integer": 1
+}
+
+PUT example-index/_doc/4
+{
+    "text" : "rrf rrf rrf rrf",
+    "integer": 2
+}
+
+PUT example-index/_doc/5
+{
+    "vector" : [0],
+    "integer": 1
+}
+
+POST example-index/_refresh
+----
+// TEST
+
+We now execute a search using RRF with a query, a kNN search, and
+a terms aggregation.
+
+[source,console]
+----
+GET example-index/_search
+{
+    "query": {
+        "term": {
+            "text": "rrf"
+        }
+    },
+    "knn": {
+        "field": "vector",
+        "query_vector": [3],
+        "k": 5,
+        "num_candidates": 5
+    },
+    "rank": {
+        "rrf": {
+            "window_size": 5,
+            "rank_constant": 1
+        }
+    },
+    "size": 3,
+    "aggs": {
+        "int_count": {
+            "terms": {
+                "field": "integer"
+            }
+        }
+    }
+}
+----
+// TEST[continued]
+
+And we receive the response with ranked `hits` and the
+terms aggregation result. Note that `_score` is replaced by
+`_rank`.
+
+[source,console-response]
+----
+{
+    "took": ...,
+    "timed_out" : false,
+    "_shards" : {
+        "total" : 1,
+        "successful" : 1,
+        "skipped" : 0,
+        "failed" : 0
+    },
+    "hits" : {
+        "total" : {
+            "value" : 5,
+            "relation" : "eq"
+        },
+        "max_score" : null,
+        "hits" : [
+            {
+                "_index" : "example-index",
+                "_id" : "3",
+                "_score" : null,
+                "_rank" : 1,
+                "_source" : {
+                    "integer" : 1,
+                    "vector" : [
+                        3
+                    ],
+                    "text" : "rrf rrf rrf"
+                }
+            },
+            {
+                "_index" : "example-index",
+                "_id" : "2",
+                "_score" : null,
+                "_rank" : 2,
+                "_source" : {
+                    "integer" : 2,
+                    "vector" : [
+                        4
+                    ],
+                    "text" : "rrf rrf"
+                }
+            },
+            {
+                "_index" : "example-index",
+                "_id" : "4",
+                "_score" : null,
+                "_rank" : 3,
+                "_source" : {
+                    "integer" : 2,
+                    "text" : "rrf rrf rrf rrf"
+                }
+            }
+        ]
+    },
+    "aggregations" : {
```

Comment:
shall we have a mention that aggs cover the union of all queries/knn or is that implicit?

Reply:
I think it's a good idea to mention. I'm thinking it may be best if I add a small follow up section on how this works as a separate PR.